[#21] Fix IllegalAccessError with StageContext

szpak opened the pull request:
More details in #21.

Fixed #21.

Commit:
[#21] Fix IllegalAccessError with StageContext

ArturGajowy commented on src/main/groovy/com/ofg/pipeline/core/PipelineBuilder.groovy:
```diff
@@ -27,7 +27,8 @@ class PipelineBuilder<P extends Project> {
 
-    private class StageContext {
+    //Cannot be private as an external closure is called in its context
+    class StageContext {
```
a test reproducing the issue would be handy ;)

szpak replied:
Good idea. I've tried it in the morning, but it doesn't want to fail when code is in jenkins-pipeline-dsl project itself. The same test case fails when run against the same codebase, however, packaged as a jar... I will try to return to that later.

Commit:
[#21] Add test reproducing issue

However, it passes even with private PipelineBuilder.StageContext when
a test is in jenkins-pipeline-dsl itself. Fails as expected whenput in
a project that uses jenkins-pipeline-dsl as a dependency...

Commit:
Remove redundant access classifier
